source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -266,8 +266,10 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     envoy::type::FractionalPercent random_sampling;
     // TODO: Random sampling historically was an integer and default to out of 10,000. We should
     // deprecate that and move to a straight fractional percent config.
+    uint64_t random_sampling_numerator{PROTOBUF_PERCENT_TO_ROUNDED_INTEGER_OR_DEFAULT(
```

Comment:
just for my sanity...   
if `tracing_config().random_sampling().value() == 100` then  This thing will return MAX_VALUE == `10000`
which would set the numerator correctly.

Reply:
Yes, see the test that I added.